roles/cfg_openwrt/templates/corerouter/config/olsrd.j2

```diff
@@ -13,6 +13,16 @@ config LoadPlugin
 	list hosts '{{ network['prefix'] | ansible.utils.ipaddr(ip_num) | ansible.utils.ipaddr('address') }} {{ host }}'
   {% endfor %}
 {% endfor %}
+{% for dhcp_static_lease in dhcp_static_leases | default([]) %}
+{% if dhcp_static_lease['name'] is defined and dhcp_static_lease['ip'] is defined and dhcp_static_lease['olsrd_nameservice'] is defined and dhcp_static_lease['olsrd_nameservice'] == 1 %}
+        list hosts '{{ dhcp_static_lease['ip'] }} {{ dhcp_static_lease['name'] }}'
```

Comment:
It might be important to make sure that no hostnames are duplicated from the assignments list.  Those are added in lines 11-15